src/Microsoft.ML.Transforms/Text/TextFeaturizingEstimator.cs

```diff
@@ -85,6 +86,27 @@ internal bool TryUnparse(StringBuilder sb)
             }
         }
 
+        /// <summary>
+        /// Defines the different type of stop words remover supported.
+        /// </summary>
+        public interface IStopWordsRemoverOptions { }
+
+        /// <summary>
+        /// Use stop words remover that can removes language-specific list of stop words (most common words) already defined in the system.
+        /// </summary>
+        public sealed class PredefinedStopWordsRemoverOptions : IStopWordsRemoverOptions { }
+
+        /// <summary>
+        /// Use stop words remover that can removes language-specific list of stop words (most common words) already defined in the system.
+        /// </summary>
+        public sealed class CustomStopWordsRemoverOptions : IStopWordsRemoverOptions
```

Comment:
>CustomStopWordsRemoverOptions [](start = 28, length = 29)

Why this interface and class is part of text transform and not part of StopWordRemover itself?

Reply:
My question is why it's not implemented same way as word ngram and char ngram implemented?

---
In reply to: [266565265](https://github.com/dotnet/machinelearning/pull/2962#discussion_r266565265) [](ancestors = 266565265)

Reply:
It can be part of StopWordRemover. I will move it.
Word ngram and char ngram are do different options that can be exclusively applied in Text transform (two parameter for two different options). On the contrary,  only one type of StopWordRemover can be applied (one parameter to take two different options).  Any idea if it can be done better than this?

---
In reply to: [266565861](https://github.com/dotnet/machinelearning/pull/2962#discussion_r266565861) [](ancestors = 266565861,266565265)